### DISCUSSION TITLE
Detect Workflow Instance Updates

### DISCUSSION THREAD

**Discussion by alirezapooneh** [elsa-workflows/elsa-core#4552] [Q&A]
Hi all,

I am going to register a callback or anything like that to detect any updates on workflow instances to fetch some data from the latest update and store it on my customized store. 
How can I do that? I use SQL server EF core persistence that may change soon to other store types.

Do I have to override the EF persistence store and do that on saving action or is there any publisher, signaler, or any service to use to detect any updates on instances?

Thanks.

**Comment by sfmskywalker** [comment]
The easiest way to achieve this is by implementing a handler for the `WorkflowInstanceSaved` notification, which is provided by the `Elsa.Workflows.Management` package.

This allows your project to store details in your customized store regardless of what persistence provider is used for Elsa.

Here's an example:

```csharp
public class WorkflowInstanceSavedHandler : INotificationHandler<WorkflowInstanceSaved>
{
    public async Task HandleAsync(WorkflowInstanceSaved notification, CancellationToken cancellationToken)
    {
        // Get information that you need from the workflow instance (accessible via the notification parameter) and store it in your own store.
    }
}
```

To register this handler, use this:

```csharp
services.AddNotificationHandler<WorkflowInstanceSavedHandler>();
```
